video_xblock/static/js/player_state.js

```diff
@@ -10,33 +10,24 @@
  * State is saved at certain events.
  */
 
-/** Run a callback when DOM is fully loaded */
-var domReady = function(callback) {
-    if (document.readyState === "interactive" || document.readyState === "complete") {
-        callback();
-    } else {
-        document.addEventListener("DOMContentLoaded", callback);
-    }
-};
+var player_state_obj = window.playerStateObj;
 var player_state = {
-    volume: {{ player_state.volume }},
-    currentTime: {{ player_state.current_time }},
-    playbackRate: {{ player_state.playback_rate }},
-    muted: {{ player_state.muted | yesno:'true,false' }},
-    transcriptsEnabled: {{ player_state.transcripts_enabled | yesno:'true,false' }},
-    captionsEnabled: {{ player_state.captions_enabled | yesno:'true,false' }},
-    captionsLanguage: '{{ player_state.captions_language }}'
+    volume: player_state_obj.volume,
+    currentTime: player_state_obj.current_time,
+    playbackRate: player_state_obj.playback_rate,
+    muted: player_state_obj.muted,
+    transcriptsEnabled: player_state_obj.transcripts_enabled,
+    captionsEnabled: player_state_obj.captions_enabled,
+    captionsLanguage: player_state_obj.captions_language
 };
 var xblockUsageId = window.location.hash.slice(1);
-var transcripts = { // eslint-disable-line
-    {% for transcript in player_state.transcripts %} // eslint-disable-line
-        '{{transcript.lang}}': { // eslint-disable-line
-            'label': '{{transcript.label}}', // eslint-disable-line
-            'url': '{{transcript.url}}', // eslint-disable-line
-        }, // eslint-disable-line
-    {% endfor %} // eslint-disable-line
-}; // eslint-disable-line
-
+var transcripts = {};
+player_state_obj.transcripts.forEach(function loopTranscript(transcript) {
```

Comment:
Do you use `loopTranscript` just for better readability the code? If no, please remove it